### PR TITLE
WT-10717 Add a reproducer to show it is possible to miss a record in the update list without WT-10522

### PR DESCRIPTION
**Pull request by etienneptl**
As WT-10522 fixes the edge case scenario, it needs to be reverted in order to fail with the suggested reproducer.

### CODE REVIEW

**Review by quchenhao — APPROVED** [review]
lgtm

File: test/suite/test_bug031.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+        # This moves the insertion @ 40 (U4) to the DS and the insertion @ 10 (U1) in the HS.
+    
+        # Read update at timestamp 10.
+        # The update chain is U4 -> U3 (aborted) -> U2 (aborted). Without the fix in WT-10522, we
+        # would exit early because of the flag WT_UPDATE_RESTORED_FROM_DS set on U2 and miss on
+        # appending U1.
+        self.session.begin_transaction('read_timestamp=' + self.timestamp_str(10))
+        cursor.set_key(key)
+        self.assertEquals(cursor.search(), 0)
```
Comment (sueloverso) [inline test/suite/test_bug031.py:146]: Doesn't the test need:
```
if __name__ == '__main__':
    wttest.run()
```

File: test/suite/test_bug031.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        evict_cursor.set_key(key)
+        evict_cursor.search()
+        evict_cursor.reset()
+        evict_cursor.close()
+
+        # This moves the insertion @ 40 (U4) to the DS and the insertion @ 10 (U1) in the HS.
+    
+        # Read update at timestamp 10.
+        # The update chain is U4 -> U3 (aborted) -> U2 (aborted). Without the fix in WT-10522, we
+        # would exit early because of the flag WT_UPDATE_RESTORED_FROM_DS set on U2 and miss on
```
Comment (sueloverso) [inline test/suite/test_bug031.py:142]: This sentence here is the entire crux of the bug. It isn't clearly called out. Please update with more details like:
Without the fix in WT-10522, the search call would return early with the XXX error instead of success because
the WT_UPDATE_RESTORED_FROM_DS flag is set on U2 and ...

File: test/suite/test_bug031.py (hunk 1 above)
Comment (etienneptl) [reply test/suite/test_bug031.py:146]: Good question. Looks like it is something we haven't followed carefully if you do:
`grep -L "wttest.run()" test/suite/test_*`

Comment (etienneptl) [reply test/suite/test_bug031.py:149]: Added it to the test in 52584dc6b5a999602bc572f850859b2abba9fc3f.
Created WT-11079 to make it consistent.

File: test/suite/test_bug031.py (hunk 2 above)
Comment (etienneptl) [reply test/suite/test_bug031.py]: Added more details in 183179f07305e57a80978edf887c5a1e28243a40. Let me know what you think.

File: test/suite/test_bug031.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    format_values = [
+        ('column', dict(key_format='r', value_format='S')),
+        ('column_fix', dict(key_format='r', value_format='8t')),
+        ('row_integer', dict(key_format='i', value_format='S')),
+    ]
+
+    scenarios = make_scenarios(format_values)
+
+    def conn_config(self):
+        config = 'debug_mode=(update_restore_evict=true)'
```
Comment (kommiharibabu) [inline test/suite/test_bug031.py:45]: I believe we do not need this config to reproduce the issue. As we are able to generate an update restore evict with uncommitted data on the page.

File: test/suite/test_bug031.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        # Since the stable timestamp is 10, RTS marks all the updates performed at a time greater
+        # time than 10 as aborted. The update chain in memory is:
+        # U3 (aborted) -> U2 (aborted) -> U1.
+        # The DS now contains the insert @ 10 (U1) and the HS is empty.
+
+        # Insert but don't commit yet.
+        self.session.begin_transaction()
+        cursor[key] = value
+
+        # Evict everything.
```
Comment (kommiharibabu) [inline test/suite/test_bug031.py:115]: How about adding more details like this eviction performs update restore evict due to the uncommitted update present on the key?

File: test/suite/test_bug031.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        evict_cursor = self.session.open_cursor(uri, None, 'debug=(release_evict)')
+        evict_cursor.set_key(key)
+        evict_cursor.search()
+        evict_cursor.reset()
+        evict_cursor.close()
+
+        # This moves the insertion @ 40 (U4) to the DS and the insertion @ 10 (U1) in the HS.
+    
+        # Read update at timestamp 10.
+        # The update chain is U4 -> U3 (aborted) -> U2 (aborted).
```
Comment (kommiharibabu) [inline test/suite/test_bug031.py:141]: This comment has two things that are discussed, 
1. Early exit of `__rec_append_orig_value`
2. Read at timestamp 10.

I prefer to move the `__rec_append_orig_value` discussion to the comment of `# This moves the insertion @ 40 (U4) to the DS and the insertion @ 10 (U1) in the HS.`

File: test/suite/test_bug031.py (hunk 2 above)
Comment (etienneptl) [reply test/suite/test_bug031.py]: Better explanation in 83634e682ce100ce7e77ea328c5cc9de72ff3eee

File: test/suite/test_bug031.py (hunk 3 above)
Comment (etienneptl) [reply test/suite/test_bug031.py]: True, removed in b901bc1efe87fa87bbc20215b6012e8fa188ccac.

File: test/suite/test_bug031.py (hunk 4 above)
Comment (etienneptl) [reply test/suite/test_bug031.py]: Added in 183179f07305e57a80978edf887c5a1e28243a40.

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: test/suite/test_bug031.py (hunk 5 above)
Comment (etienneptl) [reply test/suite/test_bug031.py]: As discussed, edited in 83634e682ce100ce7e77ea328c5cc9de72ff3eee.

**Review by sueloverso — APPROVED** [review]
LGTM

**Review by sueloverso — APPROVED** [review]
LGTM

**Review by clarissecheah — APPROVED** [review]
LGTM
